models/running_processor.go

```diff
@@ -71,34 +73,79 @@ func (r *RunningProcessor) Init() error {
 	return nil
 }
 
-func (rp *RunningProcessor) Apply(in ...telegraf.Metric) []telegraf.Metric {
-	rp.Lock()
-	defer rp.Unlock()
+func (r *RunningProcessor) ApplyFilters(metric telegraf.Metric) telegraf.Metric {
+	if ok := r.Config.Filter.Select(metric); !ok {
+		r.metricFiltered(metric)
+		return nil
+	}
 
-	ret := []telegraf.Metric{}
+	r.Config.Filter.Modify(metric)
+	if len(metric.FieldList()) == 0 {
```

Comment:
If all fields are removed then the metric needs to be dropped, I believe we are passing the metric.